src/behavior-considered-undefined.md

```diff
@@ -155,6 +127,46 @@ entire range, so it is important that the length metadata is never too large. In
 particular, the dynamic size of a Rust value (as determined by `size_of_val`)
 must never exceed `isize::MAX`.
 
+### Invalid values
+[invalid-values]: #invalid-values
+
+The Rust compiler assumes that all values produced during program execution are
+"valid", and producing an invalid value is hence immediate UB.
+
+Whether a value is valid depends on the type:
+* A [`bool`] value must be `false` (`0`) or `true` (`1`).
+* A `fn` pointer value must be non-null.
+* A `char` value must not be a surrogate (i.e., must not be in the range `0xD800..=0xDFFF`) and must be below `char::MAX`.
+* A `!` value must never exist.
+* An integer (`i*`/`u*`), floating point value (`f*`), or raw pointer must be
+  initialized, i.e., must not be obtained from [uninitialized memory][undef].
+* A `str` value is treated like `[u8]`, i.e. it must be initialized.
+* An `enum` must have a valid discriminant, and all fields of the variant indicated by that discriminant must be valid at their respective type.
+* A `struct`, tuple, and array requires all fields/elements to be valid at their respective type.
+* For a `union`, the exact validity requirements are not decided yet. The following is certain:
+  * If the `union` has a zero-sized field, then all values are valid.
+  * If a value is valid for a particular `union` field, then it is valid for the union.
+* A reference or `Box<T>` must be aligned, it cannot be [dangling], and it must point to a valid value
+  (in case of dynamically sized types, using the actual dynamic type of the
+  pointee as determined by the metadata).
+* The metadata of a wide reference, `Box<T>`, or raw pointer must match
+  the type of the unsized tail:
```

Comment:
References to `Box<T>` should be links so they'll autolink later (currently I believe we need a reference link, but these will get fixed with mdbook-spec)
```suggestion
* A reference or [`Box<T>`] must be aligned, it cannot be [dangling], and it must point to a valid value
  (in case of dynamically sized types, using the actual dynamic type of the
  pointee as determined by the metadata).
* The metadata of a wide reference, [`Box<T>`], or raw pointer must match
  the type of the unsized tail:
```